src/ui/controllers/user.ts

```diff
@@ -572,7 +575,11 @@ export async function updateProfile(
 			}
 			break
 		case 'otherAreasOfWork':
-			if (!Array.isArray(fieldValue)) {
+			response = await registry.getWithoutHal('/professions/tree')
+			if (Array.isArray(fieldValue) &&
+				fieldValue[fieldValue.length - 1] === '/professions/' + response.data.length) {
```

Comment:
I think the above logic may lead to a different result in different environments. As discussed over the call, could you please rework on it.